### PR TITLE
Issue #91 - Fix KeyError with optional NestedDictSettings in NestedDi…

### PR DESCRIPTION
**Pull request by rastytheamateur**

### CODE REVIEW

File: tests/test_appsettings.py [hunk 1]
```diff
@@ -560,6 +560,52 @@ def test_parent_setting_precedence_over_environ_value(self):
...
+            default=None,
+        )
+
+        setting.check()
+        assert setting.value is None
+
+        with override_settings(SETTING={}):
+            setting.check()
+            assert setting.value is not None
+            assert setting.value["sub"] is None
```
Comment (ziima) [inline tests/test_appsettings.py:603]: I believe `assert setting.value == {'sub': None}` would be stricter and more descriptive. Similar construction could be also used for other asserts.

Comment (rastytheamateur) [reply tests/test_appsettings.py]: Fixed.

Comment (ziima) [reply tests/test_appsettings.py]: The `assert setting.value is not None` is still there, although it duplicates the following assert.

Comment (rastytheamateur) [reply tests/test_appsettings.py]: Sorry, forgot to fix this. Now it should be done.

**Comment by rastytheamateur** [comment]
> Please squash the commits, so it can be merged.

Done.

**Pull request merged**